dlc/src/lib.rs

```diff
@@ -65,14 +68,36 @@ const ENABLE_LOCKTIME: u32 = 0xfffffffe;
 /// Represents the payouts for a unique contract outcome. Offer party represents
 /// the initiator of the contract while accept party represents the party
 /// accepting the contract.
+#[derive(PartialEq, Debug, Clone)]
 pub struct Payout {
     /// Payout for the offering party
     pub offer: u64,
     /// Payout for the accepting party
     pub accept: u64,
 }
 
+#[derive(PartialEq, Debug)]
+/// Representation of a set of contiguous outcomes that share a single payout.
+pub struct RangeOutcome {
+    /// The start of the range
+    pub start: usize,
+    /// The number of outcomes in the range
+    pub count: usize,
+    /// The payout associated with all outcomes
+    pub payout: Payout,
+}
+
+#[derive(PartialEq, Debug)]
+/// Structure that stores which stores CET and adaptor signature information.
+pub struct RangeInfo {
+    /// a cet index
+    pub cet_index: usize,
+    /// an adaptor signature index
+    pub adaptor_index: usize,
```

Comment:
At first glance I'm not actually sure I know what these indices represent?

Reply:
Tried to be a bit more descriptive, but basically the point is that CETs and adaptor signatures are kept into vectors, and so these fields store the index in those. It's a bit difficult to understand without context I must agree, but not sure how to make it clearer.